### ISSUE TITLE
Provider auto-reg: use a dedicated field to get the provider name

### ISSUE THREAD

**Issue by yugangw-msft** [Azure/azure-sdk-for-python#9272]
Please use the `target` field, and remove the regular expression matching.

```json
{
  "error": {
    "code": "MissingSubscriptionRegistration",
    "message": "The subscription registration is in 'Unregistered' state. The subscription must be registered to use namespace 'Microsoft.KeyVault'. See https://aka.ms/rps-not-found for how to register subscriptions.",
    "details": [
      {
        "code": "MissingSubscriptionRegistration",
        "target": "Microsoft.KeyVault",
        "message": "The subscription registration is in 'Unregistered' state. The subscription must be registered to use namespace 'Microsoft.KeyVault'. See https://aka.ms/rps-not-found for how to register subscriptions."
      }
    ]
  }
}
```

**Comment by RodgeFu** [comment]
This issue is create 2.5 years ago, is this still valid? Could anyone help to provide more background? Thanks.

**Comment by lmazuel** [comment]
Yes, this is still valid, it's something we didn't take to implement with @yugangw-msft. When you try to  do a call to an unregistered ARM provider, you will get an error 429 in a particular format for ARM. msrestazure (and now azure-mgmt-core) is smart enough to catch this on the fly, do a request to register the provider, and retry the request for you for free.

We complained that ARM error was tough to handle, since we had to do a regexp to extract the provider name from the error message. So ARM improved their error format to include a dedicated field with the provider name, so we don't have to use a regexp anymore. It's supposed to be more reliable. We never took the time to implement it and replace the regexp with just reading the field, since so far the regexp never failed. It's why it was low priority and still here, but one day you might get an issue if the regexp fails, and that will be because of this.
